go.mod

```diff
@@ -9,34 +11,34 @@ require (
 	github.com/blang/semver/v4 v4.0.0
 	github.com/ghodss/yaml v1.0.0
 	github.com/go-kit/log v0.2.1
-	github.com/go-openapi/strfmt v0.21.7
+	github.com/go-openapi/strfmt v0.22.0
 	github.com/google/uuid v1.6.0
 	github.com/imdario/mergo v0.3.16
-	github.com/openshift/api v0.0.0-20240327143419-1e5b8d673d96
-	github.com/openshift/client-go v0.0.0-20231218140158-47f6d749b9d9
-	github.com/openshift/library-go v0.0.0-20240216151214-738f3fa4ccf8
-	github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring v0.71.0
-	github.com/prometheus-operator/prometheus-operator/pkg/client v0.71.0
-	github.com/prometheus/alertmanager v0.26.0
+	github.com/openshift/api v0.0.0-20240618205917-987b8890c273
+	github.com/openshift/client-go v0.0.0-20240528061634-b054aa794d87
+	github.com/openshift/library-go v0.0.0-20240619120114-0c65da30ad30
+	github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring v0.74.0
+	github.com/prometheus-operator/prometheus-operator/pkg/client v0.74.0
+	github.com/prometheus/alertmanager v0.27.0
 	github.com/prometheus/client_golang v1.18.0
-	github.com/prometheus/common v0.45.0
+	github.com/prometheus/common v0.46.0
 	github.com/prometheus/prometheus v0.48.1
 	github.com/stretchr/testify v1.9.0
 	golang.org/x/sync v0.6.0
-	golang.org/x/text v0.14.0
+	golang.org/x/text v0.15.0
 	gopkg.in/yaml.v2 v2.4.0
 	gopkg.in/yaml.v3 v3.0.1
-	k8s.io/api v0.29.0
-	k8s.io/apiextensions-apiserver v0.29.0
-	k8s.io/apimachinery v0.29.0
-	k8s.io/apiserver v0.29.0
+	k8s.io/api v0.30.2
+	k8s.io/apiextensions-apiserver v0.30.1
```

Comment:
(nit) There seems to be a newer `v0.30.2` for [`k8s.io/apiextensions-apiserver`](https://pkg.go.dev/k8s.io/apiextensions-apiserver).